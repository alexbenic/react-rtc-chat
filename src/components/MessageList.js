import React from "react";
import classnames from "classnames";
import * as Types from '../types'

const MessageList = ({ current, messages }) => {
  const nickCls = classnames(
    "b",
    "mv2",
    "b2",
    "ba",
    "bw1",
    "b--light-gray",
    "pv1",
    "ph2",
    "br3"
  );

  const itemCls = message =>
    classnames("mv1", "l-list", "pt3", "bb", "bw1", "b--light-gray", {
      tr: message.sender.id === current.id
    });

  return (
    <div className="flex-auto">
      <ul className="list pl2">
        {messages.map(message => (
          <li key={message.id} className={itemCls(message)}>
            <span className={nickCls}>{message.sender.nick}</span>
            <p className="l-list f5 ph2">{message.message}</p>
          </li>
        ))}
      </ul>
    </div>
  );
};

MessageList.propTypes = {
  current: Types.User.isRequired,
  messages: Types.Messages.isRequired,
}

export default MessageList;
