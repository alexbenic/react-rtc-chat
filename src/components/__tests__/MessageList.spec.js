import React from "react";
import each from "jest-each";
import { render, cleanup } from "react-testing-library";
import randombytes from "randombytes";
import MessageList from "../MessageList";

const user = name => {
  return {
    id: randombytes(8).toString("hex"),
    nick: name,
    stream: {},
    app: 1
  };
};

const message = text => {
  return {
    id: randombytes(8).toString("hex"),
    sender: user("zvrk"),
    message: text
  };
};

afterEach(cleanup);

each([
  ["zvrk", [], 0],
  ["zoom", [message("hey"), message("you pretty boi")], 2],
  ["zoom", [message("smoooth"), message("transition"), message("babe")], 3]
]).test("user %s with messages %j", (name, messages, expected) => {
  const { container } = render(
    <MessageList current={user(name)} messages={messages} />
  );

  const list = container.querySelectorAll("li");

  expect(list.length).toBe(expected);
});

test("message list snapshot", () => {
  const { container } = render(
    <MessageList current={user("zvrk")} messages={[message("yo!")]} />
  );

  expect(container).toMatchSnapshot();
});
