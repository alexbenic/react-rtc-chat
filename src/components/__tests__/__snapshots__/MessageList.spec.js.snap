// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message list snapshot 1`] = `
<div>
  <div
    class="flex-auto"
  >
    <ul
      class="list pl2"
    >
      <li
        class="mv1 l-list pt3 bb bw1 b--light-gray"
      >
        <span
          class="b mv2 b2 ba bw1 b--light-gray pv1 ph2 br3"
        >
          zvrk
        </span>
        <p
          class="l-list f5 ph2"
        >
          yo!
        </p>
      </li>
    </ul>
  </div>
</div>
`;
